storage/innobase/xtrabackup/src/xbcloud/xbcloud.cc

```diff
@@ -696,20 +696,32 @@ bool xbcloud_put(Object_store *store, const std::string &container,
 
   xb_stream_read_done(stream);
 
-  bool ret = (res != XB_STREAM_READ_ERROR) && (!has_errors);
 
-  if (ret && upload_md5) {
-    msg_ts("%s: Uploading md5\n", my_progname);
-    ret = store->upload_object(container, backup_name + ".md5", buf_md5);
+  // check if backup directory exists and it has some files
+  if (res == XB_STREAM_READ_ERROR || has_errors ||
+      !store->list_objects_in_directory(container, backup_name, object_list) ||
+      object_list.size() == 0) {
+    msg_ts("%s: Upload failed.\n", my_progname);
+    return false;
   }
 
-  if (!ret) {
-    msg_ts("%s: Upload failed.\n", my_progname);
-  } else {
-    msg_ts("%s: Upload completed.\n", my_progname);
+  // check if xtrabackup_info, almost the last file exists in backup
+  if (std::count(object_list.begin(), object_list.end(),
+                 backup_name + "/xtrabackup_info.00000000000000000000") == 0) {
```

Comment:
why are there trailing zeros? instead of just xtrabackup_info

Reply:
pxb create files xtrabackup_info.00000000000000000000 and xtrabackup_info.00000000000000000001

Reply:
Why? I see in code the filename is defined as

  #define XTRABACKUP_INFO "xtrabackup_info"

Reply:
that's how all files in backup get into the stream. chunks of 10 MB. 
sample 
2 t1.ibd.00000000000000000000
-rw-r--r--  1 rahulmalik  staff    10M Jun 19 17:32 t1.ibd.00000000000000000001
-rw-r--r--  1 rahulmalik  staff    10M Jun 19 17:32 t1.ibd.00000000000000000002
-rw-r--r--  1 rahulmalik  staff    25B Jun 19 17:32 t1.ibd.00000000000000000005
-rw-r--r--  1 rahulmalik  staff    10M Jun 19 17:32 t1.ibd.00000000000000000003
-rw-r--r--  1 rahulmalik  staff   4.0M Jun 19 17:32 t1.ibd.00000000000000000004

Reply:
I see so we append chunk number for streams only